### PR TITLE
update namings, fix minor technical details, other polish & cleanup

### PR DESCRIPTION
**Pull request by MFatihMAR**
**Select the type of change:**

- [x] Small Changes - Typos, formatting, slight revisions
- [ ] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site and Docusaurus

**Purpose of the Pull Request:**

- `IBitWritable` → `INetworkSerializable`
- `PooledBitWriter` → `PooledNetworkWriter`
- `PooledBitReader` → `PooledNetworkReader`
- `IsReliable` bool → `Delivery` enum
- `ServerRPC` → `ServerRpc`
- `ClientRPC` → `ClientRpc`
- `BitWriter` → `NetworkWriter`
- `BitReader` → `NetworkReader`
- `BitSerializer` → `NetworkSerializer`
- `AutoBitWritable` → `AutoNetworkSerializable`
- remove pseudo-code blocks from `NetworkUpdateLoop` page
  - implementation differs from the pseudo-code we proposed on the RFC side
- use Visual Studio light & dark themes instead of default built-in themes
- fix/update minor technical details in serialization doc

**Issue Number:**

### CODE REVIEW

File: docs/advanced-topics/message-system/rpc-params.md [hunk 1]
```diff
@@ -63,11 +63,15 @@ struct ClientRpcParams
...
 [ClientRpc]
-void XyzwClientRpc(int framekey, ClientRpcParams rpcParams = default) { /* ... */ }
-```
\ No newline at end of file
+void XyzwClientRpc(int framekey, ClientRpcParams clientRpcParams = default) { /* ... */ }
+```
+
+:::tip
+`ClientRpcSendParams`'s `TargetClientIds` property is a `ulong[]` which means everytime you try to specify a subset of target clients or even a single client target, you will have to allocate a `new ulong[]`. This pattern could quickly lead into lots of heap allocations and pressure GC which would cause GC spikes at runtime. We suggest developers to cache their `ulong[]` variables or use an array pool to cycle `ulong[]` instances so that it would cause less heap allocations.
+:::
```
Comment (MFatihMAR) [inline docs/advanced-topics/message-system/rpc-params.md:77]: actually, this tip block was the main purpose of this PR but I ended up cleaning bunch of other stuff as well :)

**Review by Briancoughlin — APPROVED** [review]
Made some minor updates to a few files to reflect the edits from Fatih and files that were missing from sidebar.

**Review by lkrell — CHANGES_REQUESTED** [review]
Remove the yarn.lock edits it will break the site. Everything uses npm.
Your IDE force changed spacing in our build files (package.json and docusaurus.config.js) that I need to verify/revert.
Reviewing all other content, just putting this change in so we don't push this to any build. It will break.

**Comment by MFatihMAR** [comment]
@lkrell — I think we should update `.gitignore` so that it won't pick up `yarn.lock`.
The other changes in `package.json` and `docusaurus.config.js` files are safe though, I don't understand why we "have to" revert spacing changes.
also I wonder how can somebody check if the PR would break the build or not and also why that check is not the part of PR?

**Comment by lkrell** [comment]
I am fixing these files to clear the PR

**Comment by lkrell** [comment]
.gitignore does include the yarn.lock which is odd. No worries!

**Comment by lkrell** [comment]
Docs: Testing build. Verifying content. When passed will approve and move pr into merging.

**Comment by PioneerRedwood** [comment]
Hello, you Top developers.
Those image links are not working. Would you mind to fix it?

https://github.com/Unity-Technologies/com.unity.multiplayer.docs/blob/master/docs/advanced-topics/network-update-loop-system/network-update-loop-reference.md

**Comment by MFatihMAR** [comment]
are they really broken? @PioneerRedwood 
they're perfectly fine here:
https://docs-multiplayer.unity3d.com/docs/develop/advanced-topics/network-update-loop-system/network-update-loop-reference

**Comment by PioneerRedwood** [comment]
Oh, they broken just in readme.md file. It's OK, thank you @MFatihMAR !!

**Pull request merged**